api/scpca_portal/models/dataset.py

```diff
@@ -179,36 +179,53 @@ def files_summary(self) -> list[dict]:
             },
         ]
 
+        # cache
+        dataset_samples = self.samples
+        dataset_libraries = self.libraries
+
+        seen_samples = set()
         summaries = []
-        seen_libraries = []
 
         for file_summary_query in summary_queries:
-            if (
-                library_ids := self.libraries.filter(**file_summary_query["filter"])
-                .exclude(scpca_id__in=seen_libraries)
+            library_ids = (
+                dataset_libraries.filter(**file_summary_query["filter"])
                 .distinct()
                 .values_list("scpca_id", flat=True)
-            ):
-                nested_ids = self.original_files.filter(library_id__in=library_ids).values_list(
-                    "sample_ids", flat=True
-                )
-                samples = {s for ss in nested_ids for s in ss}
-                samples_count = len(samples)
+            )
+
+            if not library_ids:
+                continue
 
-                seen_libraries.extend(library_ids)
+            if samples_ids := (
+                dataset_samples.filter(libraries__scpca_id__in=library_ids)
+                .exclude(scpca_id__in=seen_samples)
+                .values_list("scpca_id", flat=True)
+            ):
```

Comment:
I noticed that in the sample filtering logic, we might be getting duplicate sample IDs when a sample is linked to multiple matching libraries.

For example, in the `files_summary` for `SCPCP000004`, the sample counts seem to include one extra sample:
```py
[{'samples_count': 33, 'name': 'Single-nuclei samples', 'format': '.rds'},  # should be 32?
 {'samples_count': 9, 'name': 'Single-cell samples', 'format': '.rds'}]  # should be 8?
 ```

It would be helpful to append `.distinct()` to ensure only unique samples are counted:
```suggestion

           if samples_ids := (
                dataset_samples.filter(libraries__scpca_id__in=library_ids)
                .exclude(scpca_id__in=seen_samples)
                .distinct()
                .values_list("scpca_id", flat=True)
            ):
 ```

Let me know what you think!